terraform/api/lambda.tf

```diff
@@ -0,0 +1,59 @@
+data aws_caller_identity current {}
+
+locals {
+  ecr_repository_name = "gpconnect-infra-dev-token-validation-lambda"
```

Comment:
I can't see where you created this repo. It should live in `infra` directory and the name should come from infra output. See `infra.tf` for how to get outputs from infra directory.